### PR TITLE
Only allow modifying users with donation in last 10 min

### PR DESCRIPTION
**Pull request by sedubois**

### CODE REVIEW

File: test/authorization.middleware.test.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      .catch(done);
+  });
+
+  beforeEach(() => {
+    donation = {
+      UserId: user.id,
+      currency: 'USD',
+      amount: 100
+    };
+  });
```
Comment (xdamman) [inline test/authorization.middleware.test.js]: if you just want to instantiate the `donation` variable, it doesn't have to be within a `beforeEach`.
Unless you actually meant to create the donation in the database? In which case you forgot to do `models.Donation.create(donation)`

File: test/users.routes.test.js [hunk 2]
```diff
@@ -566,35 +566,51 @@ describe('users.routes.test.js', () => {
     });
 
     it('fails if the user already has a password', done => {
```
Comment (xdamman) [inline test/users.routes.test.js]: So if the user already has a password we don't show the screen to edit its avatar/info after a donation?

**Comment by xdamman** [comment]
I made a few comments but I wanted to say that it's great to see such thorough tests! 👍

File: test/authorization.middleware.test.js (hunk 1 above)
Comment (sedubois) [inline test/authorization.middleware.test.js]: I'm just resetting the `donation` object because the tests fiddle with its `updatedAt` property.

Comment (sedubois) [inline test/authorization.middleware.test.js]: That was a leftover from previous code, will fix

File: test/users.routes.test.js (hunk 2 above)
Comment (sedubois) [inline test/users.routes.test.js]: I don't know the exact intention, but that test isn't new. I just added code to have a donation for the user to make the test pass.

**Pull request merged**